### PR TITLE
fix: convert IKEA to modern extend

### PR DESCRIPTION
**Pull request by mrskycriper**
## TODO

- [x] revise device names for consistency
- [x] fix incorrectly assigned models
- [x] sort definitions into regions
- [x] use modern extends
- [x] turn converters into modern extends
- [x] turn configurations into modern extends
- [x] move IKEA functions and converters to `lib/ikea.ts`
- [x] create zigbee2mqtt.io PR with renamed images for changed model names

## Additional
[Modern extend improvements](https://github.com/Koenkk/zigbee-herdsman-converters/pull/7239)

Partial solution for https://github.com/Koenkk/zigbee2mqtt/issues/21951

### Model changes

- `LED1923R5/LED1925G6` to `LED1923R5` (LED1925G6 already exists)
- `ICPSHC24-10EU-IL-1` to `ICPSHC24-10EU-IL-1/ICPSHC24-10EU-IL-2` (same Zigbee model)
- `ICPSHC24-30EU-IL-1` to `ICPSHC24-30EU-IL-1/ICPSHC24-30EU-IL-2` (same Zigbee model)
- `LED1934G3_E26` and `LED1934G3_E27` to `LED1934G3` (same bulbs, different sockets)
- `LED1937T5_E26` and `LED1937T5_E27` to `LED1937T5` (same bulbs, different sockets)
- `LED1903C5/LED1835C6` to `LED1835C6` (LED1903C5 is a dumb dimmable LEDARE bulb)

Images for renamed models: https://github.com/Koenkk/zigbee2mqtt.io/pull/2662

### CODE REVIEW

File: src/lib/ikea.ts [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+        const version = device.softwareBuildID.split('.').map((n) => Number(n));
+        const bindTarget = version[0] > 2 || (version[0] == 2 && version[1] > 3) || (version[0] == 2 && version[1] == 3 && version[2] >= 75) ?
+            coordinatorEndpoint : constants.defaultBindGroup;
+        await endpoint.bind('genOnOff', bindTarget);
+    };
+
+    return {configure, isModernExtend: true};
+}
+
+export function ikeaAirPurifier(): ModernExtend {
```
Comment (sjorge) [inline src/lib/ikea.ts:164]: It's probably more readable to split these into multiple modernExtends e.g.
- ikeaAirPurifierPm25
- ikeaAirPurifierFilterRunTime
- ikeaAirPurifierControlPanelLight
- ikeaAirPurifierChildLock
- ikeaAirPurifierFanSpeed
- ikeaAirPurifierFanMode

Each just having the expose, fromZigbee, toZigbee and configure that is relavent for the attribute.

Comment (mrskycriper) [reply src/lib/ikea.ts:164]: Is it necessary though? It’s not like this is going to be reused in parts anyway. Everything here comes from the same ikea purifier specific cluster

Comment (sjorge) [reply src/lib/ikea.ts:164]: Functionally it won't make much difference, just readability.

I'll probably eventually do it myself, the current bug blob of code is not very readable. The person who originally wrote it did not do a very good job at keeping it readable. (I wrote it so It's OK to say it without offending anyone)

**Comment by Koenkk** [comment]
@mrskycriper will merge it after the 1 April release.

**Pull request merged**